Solutions/Tomcat/Analytic Rules/TomcatCommandsinRequest.yaml

```diff
@@ -0,0 +1,29 @@
+id: 91f59cea-486f-11ec-81d3-0242ac130003
+name: Tomcat - Commands in URI
+description: |
+  'Detects commands in URI'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: ApacheTomcat
+    dataTypes:
+      - TomcatEvent
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1190
+  - T1133
+query: |
+  TomcatEvent
+  | where UrlOriginal contains 'cat%20/etc/passwd' or UrlOriginal contains '/etc/passwd' or UrlOriginal contains 'ping -i' or UrlOriginal contains '/usr/bin/id(' or UrlOriginal contains '%2f%75%73%72%2f%62%69%6e%2f%69%64' or UrlOriginal contains 'phpinfo()'  or UrlOriginal contains '%70%68%70%69%6e%66%6f%28%29' or UrlOriginal contains ';id' or UrlOriginal contains '%3b%69%64' or UrlOriginal contains '/bin/bash -c' or UrlOriginal contains '%2f%62%69%6e%2f%62%61%73%68%20%2d%63%27' or UrlOriginal contains '/bin/bash' or UrlOriginal contains '%2f%62%69%6e%2f%62%61%73%68' or UrlOriginal contains 'sleep(' or UrlOriginal contains '%73%6c%65%65%70%28' or UrlOriginal  contains 'curl' or UrlOriginal  contains '%63%75%72%6c' or UrlOriginal contains '&dir' or UrlOriginal  contains '%26%64%69%72' or UrlOriginal  contains '& dir' or UrlOriginal =~ '%26%20%64%69%72' or UrlOriginal contains '<script>' or UrlOriginal  contains '%3c%73%63%72%69%70%74%3e' or UrlOriginal contains 'eval(' or UrlOriginal contains '%65%76%61%6c%28' or UrlOriginal contains 'exec(' or UrlOriginal  contains '%65%78%65%63%28'  or UrlOriginal contains 'whoami' or UrlOriginal contains '%77%68%6f%61%6d%69' or UrlOriginal contains 'wget' or UrlOriginal contains 'python' or UrlOriginal contains 'gcc'  or UrlOriginal contains 'uname' or UrlOriginal contains 'systeminfo' or UrlOriginal contains '%77%67%65%74' or UrlOriginal contains '%70%79%74%68%6f%6e' or UrlOriginal contains '%75%6e%61%6d%65' or UrlOriginal  =~ '%73%79%73%74%65%6d%69%6e%66%6f' 

```

Comment:
Using CONTAINS and ORs in this way is highly intensive on the perf.  URLs are broken up by slashes, which should allow indexing for use with HAS.  
In this case, provide a list via:
`
let commands = dynamic(['cat%20/etc/passwd', '/etc/passwd', and so on]);
`
Then in body of query reference it this way:
`
| where UrlOriginal has_any (commands)
`

Reply:
fixed